### ISSUE TITLE
Consider serializing DOM objects to strings

### ISSUE THREAD

**Issue by EskelCz** [json5/json5#249]
Thank for this project, it's awesome. Except I have a corner case to figure out.
Similar to #91, I need a way to preserve information in a string, instead of throwing an error (Invalid character). In my case it's reference to a DOM node.

Example:
```
{
    target: document.body
}
```
Wanted result of parsing:
```
{
    target: 'document.body'
}
```
I know it's probably unusual but it makes perfect sense for my use case. I feel like finding those references beforehand to stringify would be similar to writing my own parsing function.

Do you think there might be a simple modification that would allow for this?
I would be ok with doing it in my own fork, it's just I'm not sure how hard would be to implement.

**Comment by jordanbtucker** [comment]
I'm not sure what you're asking. Can you explain your use case.

**Comment by EskelCz** [comment]
@jordanbtucker Simply put I have a js settings file (loaded from CDN) with an object, in which can be references to DOM elements on the page. I need to parse it from a string, which crashes on these DOM references. I’d like to keep them as strings, instead of the crash of parsing.

**Comment by HolgerJeromin** [comment]
The suggestion does not work conceptionally. 
```js
json5.stringify(document.body);
let bodyElem = document.body;
json5.stringify(bodyElem);
```
Both calls results in exact the same object as a parameter. No possibility to find "the variable name".

And serializing `target: document.body` to `target: 'document.body'` is the most surprising thing I would expect. A string with the 
`document.body.innerHTML` is a bit less surprise, but would not solve many use cases, too.

**Comment by EskelCz** [comment]
@HolgerJeromin I don't understand, the issue is with parsing a string, not stringifying an object.
It's the same as the RegExp issue, you get the same thing with stringify: `json5.stringify(/.*_tests\.js/)`. Still it's requested feature to be able to somehow retain the information when parsing it.
I know it's a hard sell, it's a corner case. My argument is simply that the error not useful, helps no-one.

Like I said, I can try to implement it on my own, was just hoping for a little guidence. Ideally an estimate of how hard would it be to do, in man-hours. Is it a one line change, or some large conceptual rework?

**Comment by HolgerJeromin** [comment]
Perhaps I mixed up the suggested direction of conversion. (I am still unsure what you mean)

How do you want to differ if the generated property `target` should get the string `'document.body'` and when to replace it with a reference to a Element in the DOM?

**Comment by EskelCz** [comment]
@HolgerJeromin Right, the reference from the current scope would be most excepted in my opinion.

But still, that would not solve my current use case. :) I'm making an editor of settings that are used in other pages. So in my case the elements would not be there, the reference would not resolve. I'm looking for a fallback that would retain these "unparsable" things as strings.

**Comment by jordanbtucker** [comment]
Thanks for the explanation. I was going to explain the complexities of how to modify this library to do what you want, but I ended up writing a prototype in the process.

https://gist.github.com/jordanbtucker/441fee538bd222cd6e9be0c061a3792e

This modification should parse simple identifiers and dot-chained identifiers as strings. For example:

```js
{ target: document }                // { target: 'document' }
{ target: document.body }           // { target: 'document.body' }
{ target: document.body.innerHTML } // { target: 'document.body.innerHTML' }
{ target: anyValidIdentifier }      // { target: 'anyValidIdentifier' }
```

It will not work on more complex, but valid, expressions like:

```js
{ target: document['body'] }
{ target: document . body }
```

If you supply a built-in JSON5 value, then it will not be parsed as a string:

```js
{ target: null }
{ target: true }
{ target: false }
{ target: Infinity }
{ target: NaN }
```

**Comment by EskelCz** [comment]
@jordanbtucker You're awesome, thank you very much!

**Comment by EskelCz** [comment]
@jordanbtucker Sorry to bother again but there's a key expression it's failing at: `document.getElementById('target')`
Would it be possible to include it as well? Selectors are really crucial.
<img width="524" alt="Screenshot 2021-07-23 at 18 33 36" src="https://user-images.githubusercontent.com/1198558/126813518-0f0c2216-7dcd-4065-9382-f45c68345718.png">

Btw I found your project JSONext, which is even more fitting to my needs (I'm using template literals), so I merged your changes there. Hopefully without any major blunder :) https://github.com/EskelCz/jsonext/blob/master/src/parse.js
